### ISSUE TITLE
Wrong entries in output

### ISSUE THREAD

**Issue by packmad** [zd2100/PScout#3]
I found entries like:

android/content/ContextWrapper,enforceUriPermission,(Landroid/net/Uri;Ljava/lang/String;Ljava/lang/String;IIILjava/lang/String;)V,Parent,5.1.1
android/content/ContentProvider,openAssetFile,(Landroid/net/Uri;Ljava/lang/String;)Landroid/content/res/AssetFileDescriptor;,Unknown,5.1.1

and a lot (more than 10k) of "DUMP" permission like:

android/net/wifi/WifiManager,isWifiEnabled,()Z,android.permission.DUMP,5.1.1

**Comment by philipphager** [comment]
@simoneaonzo any updates on this? Have you found a solution? I'm also struggling with the bulk of DUMP permissions, wrongly mapped in the current files.

**Comment by OmkarMozar** [comment]
@philipphager @six110 @zd2100  same here. I have mailed same issue to project author but no reply uptil now .. 
kindly inform me here if any updates on this issue.

**Comment by philipphager** [comment]
@six110 @OmkarMozar As I currently need valid mappings from PScout, my current solution is to used the 4.1.1 mappings from PScout 1. As the format is quite strange, I developed a parser yesterday, that converts the output to CSV with smali type support. Maybe [this](https://github.com/philipphager/pscout-parser) helps you: 

The newest mapping I was able to convert was Android 4.1.1, the output can be found [here](https://raw.githubusercontent.com/philipphager/pscout-parser/master/export/protected-methods.csv?token=AIuzK9-5NGPaxedyStaIdjUINWAXLE0Cks5YnFJQwA%3D%3D).

**Comment by OmkarMozar** [comment]
@philipphager I think latest available result for pscout 1 is API-22 . I wanted to convert it into csv .
here is the latest result available [https://github.com/zyrikby/PScout/tree/master/results](url).
Kindly let me know or update if your parser works with this result.

**Comment by philipphager** [comment]
@OmkarMozar Yes it does :)! Thank you for the link to the latest version. I uploaded the output of my converter for API 22. You can find the CSV [here](https://raw.githubusercontent.com/philipphager/pscout-parser/master/export/protected-methods.csv). Hope it'll help you!

**Comment by OmkarMozar** [comment]
@philipphager Thank you :+1: . Have you consider publishedapimapping in that folder? Those mappings seems to be missing.

**Comment by MassyB** [comment]
What does the `android.permission.DUMP` means ? is it a valide android permission, or one introduced by the tool ? if so what is it's protection level (normal, system, signature, systemOrSignature) ?

**Comment by OmkarMozar** [comment]
There is nothing called Dump permission in android. It's a bug in pscout
tool.

**Comment by MassyB** [comment]
@OmkarMozar What about the `Unkown` permission, does it mean that the API is protected by a permissions that the PScout tool isn't able to find ? or does it mean that the API is not protected at all ( another bug) ? 
There is also a `Parent` permission given by the tool.

**Comment by MassyB** [comment]
@philipphager Thanks for the provided [csv](https://raw.githubusercontent.com/philipphager/pscout-parser/master/export/protected-methods.csv) (API 22). I found some changes comparing to the original [mapping 5.1.1](http://pscout.csl.toronto.edu/data/5.1.1/mapping_5.1.1.csv) :
- when I deleted the rows containing  `android.permission.DUMP`, `Parent`, `Unknwon` as permission, I found 22 140 rows (api-permission mapping) and 96 distinct permission
- your csv contains 24 933 row (api-permission mapping) and 60 distinct permissions

Did you work on the mapping_5.1.1 ?

**Comment by philipphager** [comment]
@MassyB The csv I provided was generated using [PScout 1](http://pscout.csl.toronto.edu/). Unfortunately it does not contain all permissions and mappings of the current API, but at least the provided permissions are valid. I did not (and would not) trust the mappings produced by PScout 2, as there might not only be the problems with 'Dump', 'Parent' & 'Unknonwn',  but other invalid mappings. For my current project, I still would like to upgrade the API version of the permission mappings, so if anyone here would be interested with me to work on a revival (or fix) of PScout 2, I would be in :).

**Comment by OmkarMozar** [comment]
@philipphager I am interested. But as far I analysed pscout and android source code, It is more than fixing as whole structure got changed after Android M. Now they use java annotations to instruct the compiler that this permission is required by this method . Kindly let me know your thoughts about it.
And Yes I am in !!!

**Comment by packmad** [comment]
Now I'm using it: 
[Androguard permission mapping](https://github.com/androguard/androguard/tree/bf653d6340a05023eb7d87a70b9c7ae6327eb3cc/androguard/core/api_specific_resources/api_permission_mappings)

**Comment by OmkarMozar** [comment]
@six110  But it  still till API22. We already have it. is there something in this mapping which is not covered in pscout-1?
